### PR TITLE
Add currency entity and endpoints

### PR DESCRIPTION
**Pull request by enozcan**
Two new endpoints are added:

- `equipment/list` : returns the list of available currencies in the system. Currently: `{"JPY", "EUR", "TRY"}`
Response:
```json
{"equipments":["JPY","EUR","TRY"],"base":"USD"}
```

- `/equipment/{equipment}` : Detailed info about the currency.
Response:
```json
{"id":4,
"name":"Turkish Lira",
"code":"TRY",
"currentValue":5.7803,
"lastUpdated":1573490341000,
"timeZone":"UTC",
"valueHistory":
[{"timestamp":1563915600000,"low":5.6905,"open":5.7117,"high":5.7395,"close":5.7135},
{"timestamp":1564002000000,"low":5.6385,"open":5.7134,"high":5.775,"close":5.6992},
{"timestamp":1564088400000,"low":5.6007,"open":5.7002,"high":5.7036,"close":5.6708},
{"timestamp":1564174800000,"low":5.6708,"open":5.6708,"high":5.6708,"close":5.6708},
....
....
{"timestamp":1573333200000,"low":5.7558,"open":5.7658,"high":5.7768,"close":5.762},
{"timestamp":1573419600000,"low":5.7548,"open":5.7658,"high":5.7956,"close":5.7792}],
"currentStock":1000000.0,"predictionRate":0.1}

```
There are 3 scheduled and 1 on-startup REST calls to 3rd party service:

- `On startup`: Initializes and loads currency metadata and current values
- `after 90 sec, only once`: Initializes historical values
- `every hh:15 UTC` : update current values
- `everyday at 4:30 A.M. UTC`: update last day's historical values

### CODE REVIEW

File: backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/configuration/EquipmentConfig.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package cmpe451.group6.rest.equipment.configuration;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+
+public class EquipmentConfig {
+
+    public static final DateFormat df = new SimpleDateFormat("yyyy-MM-dd");
```
Comment (barisegesevgili) [inline backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/configuration/EquipmentConfig.java]: Is this the format that we want?

File: backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/service/EquipmentUpdateService.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+    void loadEquipmentInfo(FileInputStream is, String equipmentName){
+
+        // Load saved equipment info.
+        // Might be called on start up to prevent losing data between deployments.
+
+    }
+
+
+
```
Comment (barisegesevgili) [inline backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/service/EquipmentUpdateService.java]: Just as these lines, too many empty lines exist

File: backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/configuration/EquipmentConfig.java (hunk 1 above)
Comment (enozcan) [reply backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/configuration/EquipmentConfig.java]: This is the format the third party API service uses. Date value is stored and send back to client as timestamps

File: backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/service/EquipmentUpdateService.java (hunk 2 above)
Comment (enozcan) [reply backend/TraderX/src/main/java/cmpe451/group6/rest/equipment/service/EquipmentUpdateService.java]: You might be right. However, these were not the concern of mine when implementing service. Anyway, they're ignored in byte codes and IMO does not affect readability.

**Pull request merged**